pkg/infrastructure/services/dashboards.go

```diff
@@ -0,0 +1,147 @@
+	dashboardNames, err := fetchGrafanaDashboardNamesForURL(svcURL)
+	if err != nil {
+		return nil, err
+	}
+
+	return fetchDashboards(svcURL, dashboardNames)
+}
+
+func fetchGrafanaDashboardNamesForURL(serverURL string) ([]string, error) {
+	dashboardsURL := fmt.Sprintf("%s%s%s", serverURL, dashboardsPath, "list.json")
+	resp, err := http.Get(dashboardsURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Received NOT expected status code %d while making GET request to %s ", resp.StatusCode, dashboardsURL)
+	}
+	var dashboardNames []string
+	if err := json.NewDecoder(resp.Body).Decode(&dashboardNames); err != nil {
+		return nil, fmt.Errorf("Failed to decode response from %s into dashboard names", dashboardsURL)
+	}
+
+	return dashboardNames, nil
+}
+
+func fetchDashboards(serverURL string, dashboardNames []string) ([]GrafanaDashboard, error) {
+	var dashboards []GrafanaDashboard
+	for _, name := range dashboardNames {
+		dashboardURL := fmt.Sprintf("%s%s%s", serverURL, dashboardsPath, name)
+		resp, err := http.Get(dashboardURL)
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		if resp.StatusCode == http.StatusNotFound {
+			return nil, nil
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("Received NOT expected status code %d while making GET request to %s ", resp.StatusCode, dashboardURL)
+		}
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Fatal(err)
+		}
+		dashboards = append(dashboards, GrafanaDashboard{Name: name, RawJSONDashboard: string(bodyBytes)})
+	}
+	return dashboards, nil
+}
+
+func configureGrafanaDashboards(client *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme, namespace string) (time.Duration, error) {
+	dashboards, err := fetchGrafanaDashboards(client, kogitoService)
+	if err != nil {
+		return reconciliationPeriodAfterDashboardsError, err
+	}
+
+	reconcileAfter, err := deployGrafanaDashboards(dashboards, client, kogitoService, scheme, namespace)
+
+	return reconcileAfter, err
+}
+
+func deployGrafanaDashboards(dashboards []GrafanaDashboard, cli *client.Client, kogitoService v1alpha1.KogitoService, scheme *runtime.Scheme, namespace string) (time.Duration, error) {
+	for _, dashboard := range dashboards {
+		resourceName := strings.ReplaceAll(strings.ToLower(dashboard.Name), ".json", "")
+		dashboardDefinition := &grafanav1.GrafanaDashboard{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: namespace,
+				Labels: map[string]string{
+					framework.LabelAppKey: GrafanaDashboardAppName,
+				},
+			},
+			Spec: grafanav1.GrafanaDashboardSpec{
+				Json: dashboard.RawJSONDashboard,
+				Name: dashboard.Name,
+			},
+		}
+		if _, err := kubernetes.ResourceC(cli).CreateIfNotExistsForOwner(dashboardDefinition, kogitoService, scheme); err != nil {
```

Comment:
Minor - this call returns boolean parameter indicating whether dashboard was created. It would be nice to show `Successfully created grafana dashboard` below just in case object was created (to keep operator log clean)